### PR TITLE
Remove automatic resource/scope handling

### PR DESCRIPTION
**Pull request by PinpointTownes**
Note: I removed the automatic resource/scope handling but I kept the extra checks that ensure a client application doesn't ask for more scopes/resources than originally granted by the authorization server.

Also, per @sunsided's suggestion, the "openid" scope is automatically included if the developer doesn't call `SetScopes` (except when the request is not an OIDC request).

/cc @damccull @sunsided @DovydasNavickas

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Snif, I'll miss you so much, lovely feature :sob:

**Comment by DovydasNavickas** [comment]
As you said:
```csharp
ticket.SetResources(request.GetResources());
ticket.SetScopes(request.GetScopes());
```
:smile:

**Comment by damccull** [comment]
Soooo... What's it do in layman's terms?

**Comment by PinpointTownes** [comment]
@damccull with this change, adding `resource=resource_server` in your OAuth2/OIDC request will now have **no effect** on the audience of the access token. It's not a big deal if you're using opaque tokens and the validation middleware (since the audience is not checked by default, unless you set the `Audience` property in the validation options), but will cause errors when using the JWT bearer middleware, because the JWT access token won't have an `aud` claim.

To "fix" that, you need to use `SetResources` to add the resource/audience in the authentication ticket:

```csharp
ticket.SetResources(new[] { "resource_server" });
```

The **same change applies to `scope`**: the `scope` parameter of the OAuth2/OIDC request no longer controls the granted scope (i.e the `scope` that is persisted in the access token), it's just a hint containing the "scopes" the client app would like to get.

For instance, if you add `scope=openid offline_access` to your OIDC parameters but don't call `ticket.SetScopes(new[] { "openid", "offline_access" })` in your own code, you won't get any refresh token in the OIDC response.

**Comment by DovydasNavickas** [comment]
Actually, from "offline_access" point of view it's an awesome change. Requesting refresh tokens "just because you can" is a way clients were going quite usually. So it's a +1 from me just for that. And of course other positive outcomes.

**Comment by sunsided** [comment]
Thumbs up as well. :+1:

**Comment by PinpointTownes** [comment]
@DovydasNavickas @sunsided glad you like this change :smile:
